geode-core/src/main/java/org/apache/geode/internal/security/IntegratedSecurityService.java

```diff
@@ -125,6 +126,20 @@ public Subject getSubject() {
     return currentUser;
   }
 
+  @Override
+  public Object getPrincipal() {
+    Subject subject;
+    try {
+      subject = getSubject();
+    } catch (AuthenticationRequiredException ignored) {
```

Comment:
I think we should not catch this exception. When this happens, something is seriously wrong and nothing should behave correctly afterwards.

Reply:
Fixed